crates/snapshot/src/snapshotter.rs

```diff
@@ -17,6 +17,10 @@ pub type SnapshotterResult = Result<SnapshotTargets, SnapshotterError>;
 /// The snapshotter type itself with the result of [Snapshotter::run]
 pub type SnapshotterWithResult<DB> = (Snapshotter<DB>, SnapshotterResult);
 
+/// Snapshots are initially created in `{...}/datadir/snapshots/temp` and moved once finished. This
+/// directory is cleaned up on every booting up of the node.
+const TEMPORARY_SUBDIRECTORY: &str = "temp";
```

Comment:
why not system-wide `std::env::temp_dir`?

Reply:
that would go into `/tmp/`, which the user might not be aware and might not have storage space for it and/or might be in a different partition.

With it being inside the datadir, we're at least sure that there is space